src/custom/components/CowSubsidyModal/utils.test.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @jest-environment ./custom-test-env.js
+ */
+import { BigNumber } from 'bignumber.js'
+import { getDiscountFromBalance } from './utils'
+
+const MOCK_TIER_DATA: { [threshold: string]: number } = {
+  '0': 0,
+  '100': 10,
+  '1000': 15,
+  '10000': 25,
+  '100000': 35,
+}
+
+const MOCK_BALANCES = [0, 150, 3512, 64884, 5501230]
+const FALSE_MOCK_BALANCES = [100, 99, 10002, 12, 0]
```

Comment:
i find strange how u structured this. 

basically there's some coupling between `MOCK_TIER_DATA` and `MOCK_BALANCES` and `FALSE_MOCK_BALANCES`, and i only understood after reading the code which takes a bit of energy (describe inside describe with an iterator inside another iterator and and `it` following`). So u did it so the position of the elements in the arrays refers to the tier they should be in.

I find even more strange the `FALSE_MOCK_BALANCES`.

Wouldn't be easier to define your cases like for example:

```js
BALANCES_IN_TIER = [
   { balance: '0', tier: 0 },
   { balance: '0.1', tier: 1 },
   { balance: '99.999', tier: 1 },
   { balance: '100', tier: 2 },
   { balance: '150', tier: 2 },
   ...
]
```

This way is easier to read, add tests and reason about imo.

Reply:
you're also code reviewing at like 23h 
i can re-review this later

Reply:
accidentally pushed these recommended changes to the follow up PR #2558 

here is the commit with the changes: [39052f9](https://github.com/gnosis/cowswap/pull/2558/commits/39052f9dd558caae3035d675a4c39a04ae4752c6)

Reply:
Reviewed! closing here then